pvlib/iotools/pvgis.py

```diff
@@ -516,17 +522,16 @@ def get_pvgis_tmy(latitude, longitude, outputformat='json', usehorizon=True,
     elif outputformat == 'csv':
         with io.BytesIO(res.content) as src:
             data, months_selected, inputs, meta = _parse_pvgis_tmy_csv(src)
-    elif outputformat == 'basic':
-        with io.BytesIO(res.content) as src:
-            data, months_selected, inputs, meta = _parse_pvgis_tmy_basic(src)
     elif outputformat == 'epw':
         with io.StringIO(res.content.decode('utf-8')) as src:
             data, meta = parse_epw(src)
             months_selected, inputs = None, None
+    # raise exception if pvgis format isn't in ['csv', 'json', 'epw']
     else:
-        # this line is never reached because if outputformat is not valid then
-        # the response is HTTP/1.1 400 BAD REQUEST which is handled earlier
-        pass
+        err_msg = (
+            "pvgis format '{:s}' was unknown, must be either 'json', 'csv', or"
+            " 'epw'.").format(outputformat)
+        raise ValueError(err_msg)
```

Comment:
move the condition for `'basic'` and changes from `read_pvgis_hourly()` to here:
```
elif outputformat == 'basic':
    err_msg = ("outputformat='basic' is no longer supported, please use ",
                      "outputformat='csv' instead.")
    raise ValueError(err_msg)